tests/zep/test_postgres_datasource.py

```diff
@@ -145,9 +120,10 @@ def test_construct_table_asset_directly_with_splitter():
         asset = postgres_datasource.TableAsset(
             name="my_asset",
             table_name="my_table",
-            datasource=source,
             column_splitter=splitter,
         )
+        # TODO: asset custom init
+        asset._datasource = source
```

Comment:
Is this a consequence of having a combined runtime/config model?

Reply:
Yes, it causes recursion errors when added to the constructor as if it were a normal field.
We can use a custom init method or a conditional in the look hook methods to do deal with it as well.
But considering we may be moving to a `2 model` design, I didn't think it was worth that extra complication yet.